tasks/typescript/hard/main.ts

```diff
@@ -10,35 +8,51 @@ interface Product {
 let products: Product[] = [];
 
 async function fetchData() {
-    // TODO: Implement the fetch function
-    displayProducts(products);
+    try {
+        const response = await fetch('https://dummyjson.com/products');
+        const data = await response.json();
```

Comment:
Code looks really good! It is generally good idea to check if the response is actually ok with something like this:
```bash
if (!response.ok) {
    throw new Error(`HTTP error! status: ${response.status}`);
}
```
Otherwise, i don't really see any major things that could be improved here, nicely done!

Reply:
yeah, next time i'll take care.
Thank you!